default/scripting/macros/misc.py

```diff
@@ -68,14 +68,17 @@ def DESCRIPTION_EFFECTSGROUP_MACRO(desc: str):
         1
         + StatisticIf(float, condition=IsTarget & EmpireHasAdoptedPolicy(name="PLC_AUGMENTATION"))
         # slower growth with augmentation on low-infrastructure planets
-        * MinOf(
-            float,
-            1.2,
-            MaxOf(
+        * (
+            MinOf(
                 float,
-                0.5,
-                Target.Construction / NamedRealLookup(name="AUGMENTATION_FULL_GROWTH_INFRASTRUCTURE_REQUIREMENT"),
-            ),
+                1.2,
+                MaxOf(
+                    float,
+                    0.5,
+                    Target.Construction / NamedRealLookup(name="AUGMENTATION_FULL_GROWTH_INFRASTRUCTURE_REQUIREMENT"),
+                ),
+            )
+            - 1
```

Comment:
this looks wrong. it should be `(1+min(1.2, max-1))` and not `(1+(min(1.2, max)-1))`

Reply:
As explanation: the min is the cutoff for the maximum effect, so it should be probably outside as far as possible. 1.2 would giving 120% max effect,  so in the end a bit more than doubling. If that is too high, the value could be lower. Any way it should be consistent with the focs.txt

Putting it another level outside would be even clearer code for the max growth, but worse in computation (because it always happens ; also if the policy is not adopted).

Reply:
Note as a bonus that you could also define a named real with the resulting minimum and maximum multipliers for showing in the pedia. So in this case the minimum being 0.5 and the maximum being 2.2

Reply:
I'm not understanding something here. The -1 is outside the min/max function to turn a range of 1.5 - 2.2 into a range of 0.5 - 1.2. Putting it inside the min/max function would keep the same range of 1.5 - 2.2 just increase the infrastructure needed to hit 2.2?

Reply:
ok, maybe you are looking at turning it to a range of 0.5 - 2.2? in that case both the focs and py would be wrong

Reply:
Just speaking balancing wise 0.5 -2.2 would at least make the policy usable, Should I modify the fix to aim for range 0.5 - 2x

Reply:
i was not commenting on balancing, personally I would be OK with 0.5-2.2 (having a noticable effect seems to be better than having a neglible effect)

i was commenting that the python and text focs are using different brackets / are using inconsistent implementations.

intention was to substitute the cyborg tech effect with a policy in 141e4cbe8c775b4f71dcd036e8d31f7dc7bbc08f (also quickfix 65deee6c12db0ff2136caf59da8676fa0ef32400 ). I think there was no growth rate adjument with the cyborg tech

Reply:
ok double checked. You are right that they don't match. The python is the intended change:

growth rate = 1 + ( (if policy) * ( (range of 0.5 to 1.2) - 1 ) )

Reply:
Perhaps the brackets are less confusing if the equation is in a separate macro?

So: * 1 + ((if policy) * (augmentation_modifier)) ? With augmentation_modifier being a range of -0.5 to +1?

Reply:
updated, but need to test. Is there a way to test the .focs and .py separately?

Reply:
i guess we should check if the focs.txt version is actually in use. Else I we could drop that. If it is not in use, one could write some focs.txt content to test it. But that seems rather artificial
there was also some facility from o01eg to compare if the resulting c++ structures are the same - in that case only one version would need to be tested.

Reply:
currently only the focs.py is in use
```
$ grep -r GROWTH_RATE_FACTOR default/scripting/
default/scripting/macros/misc.py:GROWTH_RATE_FACTOR = (
default/scripting/macros/misc.macros:GROWTH_RATE_FACTOR
default/scripting/techs/ship_hulls/SHP_GAL_EXPLO.focs.py:from macros.misc import GROWTH_RATE_FACTOR, MIN_RECOLONIZING_HAPPINESS, MIN_RECOLONIZING_SIZE
default/scripting/techs/ship_hulls/SHP_GAL_EXPLO.focs.py:                * GROWTH_RATE_FACTOR
default/scripting/species/species_macros/population.py:from macros.misc import GROWTH_RATE_FACTOR
```